### PR TITLE
fix(core): static-query migration should gracefully exit if AOT compiler throws

### PR DESCRIPTION
**Pull request by devversion**
* See individual commits (except first one)

**Note**: Blocked on #30254

### CODE REVIEW

**Review by CaerusKaru — APPROVED** [review]
LGTM

**Review by filipesilva — APPROVED** [review]

File: packages/core/schematics/test/static_queries_migration_template_spec.ts [hunk 1]
```diff
@@ -457,17 +461,55 @@ describe('static-queries migration with template strategy', () => {
...
+        @NgModule({declarations: [MyComp]})
+        export class MyModule {}
+      `);
+
+      writeFile('/components.ts', `export * from './my-component';`);
+      writeFile('/index.ts', `export * from './app-module';`);
+
+      // Enable flat-module bundling in order to simulate a common AOT compiler
+      // failure that can happen in CLI projects that use flat-module bundling
+      // e.g. with ng-packager. https://github.com/angular/angular/issues/20931
```
Comment (kara) [inline packages/core/schematics/test/static_queries_migration_template_spec.ts]:
```suggestion
      // e.g. with ng-packagr. https://github.com/angular/angular/issues/20931
```

**Review by kara — APPROVED** [review]
LGTM, found a typo though

**Comment by kara** [comment]
@devversion Also could you rename the commit to `fix(core)`? Refactors technically don't change behavior.

File: packages/core/schematics/test/static_queries_migration_template_spec.ts (hunk 1 above)
Comment (devversion) [reply packages/core/schematics/test/static_queries_migration_template_spec.ts]: Done.

**Comment by devversion** [comment]
@kara Done. I wasn't too sure if we want to list these changes in the changelog.

**Comment by alxhub** [comment]
[Presubmit](https://test.corp.google.com/ui#id=OCL:247237439:BASE:247237476:1557336102801:265e8b89)

**Pull request closed without merging**